K_TIMER_DEFINE macro causing build error

@andyross Commit 150e18de6e18069426361fe30150bd31ee5250db appears to be causing a build error.

K_TIMER_DEFINE appears to be causing a build error in 2.3:
```
K_TIMER_DEFINE(heartbeat_timer, heartbeat_timer_handler, NULL);
```
```
/Users/stevenslupsky/Documents/source/zephyrproject/zephyr/include/kernel.h:1785:2: error: designator order for field '_timeout::fn' does not match declaration order in '_timeout'
  }
  ^
/Users/stevenslupsky/Documents/source/zephyrproject/zephyr/include/kernel.h:1841:3: note: in expansion of macro 'Z_TIMER_INITIALIZER'
   Z_TIMER_INITIALIZER(name, expiry_fn, stop_fn)
   ^~~~~~~~~~~~~~~~~~~
../src/main.cpp:294:1: note: in expansion of macro 'K_TIMER_DEFINE'
 K_TIMER_DEFINE(heartbeat_timer, heartbeat_timer_handler, NULL);
 ^~~~~~~~~~~~~~

```

The problem is here:
```
#define Z_TIMER_INITIALIZER(obj, expiry, stop) \
	{ \
	.timeout = { \
		.node = {},\
		.dticks = 0, \
		.fn = z_timer_expiration_handler \
	}, \
	.wait_q = Z_WAIT_Q_INIT(&obj.wait_q), \
	.expiry_fn = expiry, \
	.stop_fn = stop, \
	.status = 0, \
	.user_data = 0, \
	_OBJECT_TRACING_INIT \
	}
```

and here:
```
struct _timeout {
	sys_dnode_t node;
	_timeout_func_t fn;
#ifdef CONFIG_TIMEOUT_64BIT
	/* Can't use k_ticks_t for header dependency reasons */
	uint64_t dticks;
#else
	uint32_t dticks;
#endif
};
```

The order of the struct members was changed and the macro was not updated to reflect this.

--- a/include/kernel.h
+++ b/include/kernel.h
@@ -1773,8 +1773,8 @@ struct k_timer {
 	{ \
 	.timeout = { \
 		.node = {},\
+		.fn = z_timer_expiration_handler, \
 		.dticks = 0, \
-		.fn = z_timer_expiration_handler \
 	}, \
 	.wait_q = Z_WAIT_Q_INIT(&obj.wait_q), \
 	.expiry_fn = expiry, \

